### DISCUSSION TITLE
Deprecated QueryExecutors constructor

### DISCUSSION THREAD

**Discussion by VascoTacos** [graphql-java-generator/graphql-maven-plugin-project#146] [Q&A]
I see that the following constructor has been marked as deprecated:

`public QueryExecutor(String graphqlEndpoint, Client client)`
  
Would you mind explaining why is has been marked as deprecated? 

Main reason why I ask is because I cant figure out a way to log the actual body of the request right before it is sent over the wire using WebClient I can with the a Client object.

Ex log:
![Screen Shot 2022-07-22 at 1 21 17 PM](https://user-images.githubusercontent.com/84336277/180492071-84218852-b24b-46ab-918c-c7f0fbcf7ae2.png)

**Comment by etienne-sf** [comment]
There are two points here, right :
* the deprecation of QueryExecutor constructor. 
* the log of the http query sent to the server. 

I don't understand the link between both. 

About the first one: in v2.x version, the Query, Mutation and Subscription will be spring beans. So only default constructors will remain. 

About the second one: change the log configuration, and put the com.graphql-Java-generator log level. It should do what you're expecting. 

Étienne

**Reply by VascoTacos** [reply]
The link between the two: 

The deprecated QueryExecutor constructor accepts a Client object. The Client object allows for configuration of the log level as so:

```
        ClientConfig clientConfig = new ClientConfig();
        clientConfig.register( new LoggingFeature( java.util.logging.Logger.getLogger( LoggingFeature.DEFAULT_LOGGER_NAME ),
                                                   Level.INFO,
                                                   LoggingFeature.Verbosity.PAYLOAD_ANY,
                                                   51200 ) );
        clientConfig.property( "jersey.config.logging.verbosity", LoggingFeature.Verbosity.PAYLOAD_ANY );
        java.util.logging.Logger.getLogger( LoggingFeature.DEFAULT_LOGGER_NAME ).setLevel( Level.ALL );

        QueryExecutor query = new QueryExecutor( "http://localhost:8080/query", ClientBuilder.newClient( clientConfig ) );
```

But now I know why it is marked as deprecated. 


> About the second one: change the log configuration, and put the com.graphql-Java-generator log level. It should do what you're expecting.

You mean set log level to debug? If so, I did see request logs but I didn't see any logging from the actual client right before it is sent over the wire. The logging I was able to see came from QueryExecutor.

**Reply by etienne-sf** [reply]
Are you talking about the javax.ws.rs.client.Client ?

Then it's another subject. 

If you use this constructor, then you provide this client : so you'll have to enable the logging for the class you provided, which is out of topic of this plug-in. 

Étienne
